src/http.cc

```diff
@@ -2193,6 +2193,29 @@ HttpStateData::buildRequestPrefix(MemBuf * mb)
     return mb->size - offset;
 }
 
+void
+HttpStateData::refreshConnectionCloseFlag()
+{
+    if (request->flags.mustKeepalive)
+        flags.connectionClose = false;
+    else if (request->flags.pinned)
+        flags.connectionClose = !request->persistent();
+    else if (!Config.onoff.server_pconns)
+        flags.connectionClose = true;
+    else if (flags.tunneling)
+        // tunneled non pinned bumped requests must not keepalive
+        flags.connectionClose = request->flags.sslBumped;
+    else if (!_peer)
+        flags.connectionClose = false;
+    else if (_peer->stats.n_keepalives_sent < 10)
+        flags.connectionClose = false;
+    else if ((double) _peer->stats.n_keepalives_recv /
+             (double) _peer->stats.n_keepalives_sent > 0.50)
```

Comment:
Please use proper C++ casts and `{}` here even though you are just copying this code.